src/main/java/uk/ac/cam/cl/dtg/segue/scheduler/SegueJobService.java

```diff
@@ -142,13 +142,18 @@ public void registerScheduleJob(final SegueScheduledJob jobToRegister) throws Sc
 
         scheduler.getContext().put(jobToRegister.getExecutableTask().getClass().getName(), jobToRegister.getExecutionContext());
 
-        if (!scheduler.checkExists(job.getKey())) {
+        if (!scheduler.checkExists(job.getKey()) || !scheduler.checkExists(cronTrigger.getKey())) {
             scheduler.scheduleJob(job, cronTrigger);
-            log.info(String.format("Registered job (%s) to segue job execution service. Current jobs registered (%s): ", jobToRegister.getJobKey(), localRegisteredJobs.size()));
+            log.info("Registered Quartz job ({}). Current jobs registered ({}): ", jobToRegister.getJobKey(), localRegisteredJobs.size());
         } else {
-            // FIXME - this does not update the job details, e.g. if the SQL file name changes.
-            scheduler.rescheduleJob(cronTrigger.getKey(), cronTrigger);
-            log.info(String.format("Re-registered job (%s) to segue job execution service. Current jobs registered (%s): ", jobToRegister.getJobKey(), localRegisteredJobs.size()));
+            CronTrigger existingTrigger = (CronTrigger) scheduler.getTrigger(cronTrigger.getKey());
+            if (!existingTrigger.equals(cronTrigger)) {
```

Comment:
I changed the `cronString` of a job already registered, but `existingTrigger.equals(cronTrigger)` still resolves `true`, so I don't think this is working as expected.

Reply:
Thanks for spotting this! 🙌